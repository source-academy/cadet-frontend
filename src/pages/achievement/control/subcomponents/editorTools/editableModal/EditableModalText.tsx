import { EditableText } from '@blueprintjs/core';
import React from 'react';

type EditableModalTextProps = {
  goalText: string;
  setGoalText: any;
};

function EditableModalText(props: EditableModalTextProps) {
  const { goalText, setGoalText } = props;

  return (
    <>
      <EditableText
        placeholder={`Enter your goal text here`}
        value={goalText}
        onChange={setGoalText}
        multiline={true}
      />
    </>
  );
}

export default EditableModalText;
