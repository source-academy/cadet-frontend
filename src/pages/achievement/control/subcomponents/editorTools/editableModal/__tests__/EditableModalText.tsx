import { mount } from 'enzyme';
import React from 'react';

import EditableModalText from '../EditableModalText';

const mockProps = {
  goalText: 'Goal Text',
  setGoalText: () => {}
};

test('EditableModalImage component renders correctly', () => {
  const text = <EditableModalText {...mockProps} />;
  const tree = mount(text);
  expect(tree.debug()).toMatchSnapshot();
});
