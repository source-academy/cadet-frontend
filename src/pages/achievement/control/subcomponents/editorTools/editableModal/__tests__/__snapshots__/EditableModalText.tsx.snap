// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableModalImage component renders correctly 1`] = `
"<EditableModalText goalText=\\"Goal Text\\" setGoalText={[Function: setGoalText]}>
  <Blueprint3.EditableText placeholder=\\"Enter your goal text here\\" value=\\"Goal Text\\" onChange={[Function: setGoalText]} multiline={true} alwaysRenderInput={false} confirmOnEnterKey={false} defaultValue=\\"\\" disabled={false} maxLines={Infinity} minLines={1} minWidth={80} type=\\"text\\">
    <div className=\\"bp3-editable-text bp3-multiline\\" onFocus={[Function (anonymous)]} tabIndex={0}>
      <span className=\\"bp3-editable-text-content\\" style={{...}}>
        Goal Text
      </span>
    </div>
  </Blueprint3.EditableText>
</EditableModalText>"
`;
